### PR TITLE
merge Documentation/template with master

### PR DESCRIPTION
**Pull request by elswit**
Summary:

* Updated CONTRIBUTING.md with step-by-step information for adding new test problem
* Converted the .md to .rst and integrated it in the docs

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,50 +1,161 @@
...
+2. Create a new folder in the `src/+opt/` directory. Follow the same naming conventions as the existing problems. Start the name with `+` to maintain the structure of the Matlab/Octave package. Also create a subfolder named `+presets` in the new problem folder:
+
+```bash
+mkdir src/+opt/+newtest
+mkdir src/+opt/+newtest/+presets
+cd src/+opt/+newtest/
+```
+
+3. The minimal set of files needed inside the problem folder to set up a new test problem are:
+  *  The right-hand-side function named as `f.m`
```
Comment (AndreyAPopov) [inline CONTRIBUTING.md:29]: I feel like explicitlly give commands to create those files, so either do vim/nano, or touch or maybe the way to open with matlab?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,50 +1,161 @@
...
-A problem package must also contain a class named `<Name>Parameters.m`. It only needs to provide public properties for
-each of the problem parameters; no constructor or methods are needed. Property validation is currently not supported in
-Octave. Therefore, we use a custom comment syntax that is parsed by the installer to optionally include validation.
+```Matlab
 
-```matlab
-% Example
-properties
-    MyProp %MATLAB ONLY: (1,1) {mustBeInteger, mustBePositive}
+classdef SimpleProblem < otp.Problem
```
Comment (AndreyAPopov) [inline CONTRIBUTING.md:58]: There is an inconsistency here. You named the directory "+newtest", but the problem "SimpleProblem". I feel like change this to "NewTestProblem"

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,50 +1,161 @@
...
+
+    methods
+        function obj = SimpleProblem(timeSpan, y0, parameters)
+            obj@otp.Problem('Simple Test Problem', [], timeSpan, y0, parameters);
```
Comment (AndreyAPopov) [inline CONTRIBUTING.md:62]: ('New Test', []

I think we agreed to not have 'Problem' In this internal name.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,50 +1,161 @@
...
+    end
+    
+    methods (Access=protected)        
+        function onSettingsChanged(obj)
+
+            % parameters are stored in the obj.Parameters structure 
+            % We can assign them to individual variables 
+            % to be used in function calls
+
+            param1 = obj.Parameters.param1; % ...
```
Comment (AndreyAPopov) [inline CONTRIBUTING.md:73]: As param1 is a property, it should be Param1 (capitalized).

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -1,50 +1,161 @@
...
-descriptor at the end, e.g., `jacobianStiff.m`.
 
-### Presets
+```Matlab
+
+classdef SimpleParameters
+
+    %SimpleParameters
+    properties
+        param1 %MATLAB ONLY: (1,1) {mustBeNumeric, mustBeReal, mustBeNonnegative}
```
Comment (AndreyAPopov) [inline CONTRIBUTING.md:114]: ```Param1```

**Comment by elswit** [comment]
Need to go ahead with this merge so that we can create (permanent) links to appropriate sections of the documentation in the draft of the paper

**Pull request merged**